Assets/MixedRealityToolkit-SDK/Features/Audio/Influencers/AudioInfluencerController.cs

```diff
@@ -0,0 +1,313 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.Audio.Influencers
+{
+    /// <summary>
+    /// Class which supports components implementing <see cref="IAudioInfluencer"/> being used with audio sources.
+    /// </summary>
+    /// <remarks>
+    /// AudioInfluencerController requires an <see cref="AudioSource"/> component. If one is not attached, it will be added automatically.
+    /// Each sound playing game object needs to have an AudioInfluencerController attached in order to have it's audio influenced.
+    /// </remarks>
+    [RequireComponent(typeof(AudioSource))]
+    [DisallowMultipleComponent]
+    public class AudioInfluencerController : MonoBehaviour
```

Comment:
For some reason I'm getting a strong vibe that we should be making an interface for this and putting it in core so people can implement their own kinds of influencers.

Reply:
they are SDK constructs folks already CAN implement their own influencers. and the controller has to be a monobehavior

Reply:
Let's move it into core and use the event system. We should stick to our patterns we've set for the MRTK and not worry about how we've done things in HTK.

Reply:
this is not intended to be a core system it never has been.

Reply:
the event system does not fit this scenario.

Reply:
I disagree, as we're using an interface and using the event data would help keep us from having to get components (which is rather costly)

Reply:
costly when done tons of times every frame. at a human scale (4 times per second) it is not that expensive

Reply:
I suppose we'll have to see when the GC kicks in and we get stalls on HL or another mobile device. 

In either case, we should at least take a look at what this might look like instead of dismissing it bc it's different than the way it was originally written. Remember the MRTK is not the HTK.